test/e2e/data/infrastructure-docker/v1beta1/main/bases/cluster-with-kcp.yaml

```diff
@@ -85,6 +88,8 @@ spec:
         # host.docker.internal is required by kubetest when running on MacOS because of the way ports are proxied.
         certSANs: [localhost, 127.0.0.1, 0.0.0.0, host.docker.internal]
     initConfiguration:
+      localAPIEndpoint:
+        bindPort: 7777
```

Comment:
This confuses me a little bit.
I was assuming we want to make the CP port configurable, which in my mind is the port the LB balancer frontend answer to. But this instead is changing the port the CP backends answer to, which is a layer down the stack 🤔

Reply:
Ah good point. The DockerCluster APIEndpoint port is definitely the external port on the lb.

Not sure if we should also modify the port the apiserver is listening on.

I just remember that the whole situation around the apiserver port is messy and differently implemented across infra providers (if I understood it correctly). xref: https://github.com/kubernetes-sigs/cluster-api/pull/6272

Maybe we should only modify the lb port here and just keep the apiserver port the same to keep it simpler?

Reply:
+1 to keep the apiserver port the same to keep it simpler (unless there is a strong requirement to make it configurable)